File: CPlusPlus/CppLibrary/StringBuilder.cpp
#include "StringBuilder.h"

/**
 * Constructs a string builder with no characters in it and an initial
 * capacity of 16 characters.
 */
StringBuilder::StringBuilder() {
    // no code yet
}

/**
 * Appends the specified string to this character sequence.
 *
 * @param str a string.
 * @return a pointer to this object.
 */
StringBuilder *StringBuilder::append(String *str) {
    oss << str->_s();
    
    return this;
}

/**
 * Returns a string representing the data in this sequence.
 *
 * @return a string representation of this sequence of characters.
 */
String *StringBuilder::toString() {
    return new String(oss.str());
}

File: CSharp/CsLibrary/StringBuilder.cs
﻿namespace JavaClasses
{

    /// <summary>
    /// A mutable sequence of characters.
    /// </summary>
    /// <author>Petr Kozler (A13B0359P)</author>
    public class StringBuilder : Object
    {

        // an inner string builder
        private System.Text.StringBuilder sb;

        /// <summary>
        /// Constructs a string builder with no characters in it and an initial
        /// capacity of 16 characters.
        /// </summary>
        public StringBuilder()
        {
            sb = new System.Text.StringBuilder();
        }

        /// <summary>
        /// Appends the specified string to this character sequence.
        /// </summary>
        /// <param name="str">a string.
        /// </param><returns>a reference to this object.
        /// </returns>
        public StringBuilder append(String str)
        {
            sb.Append(str.ToString());

            return this;
        }

        /// <summary>
        /// Returns a string representing the data in this sequence.
        /// </summary>
        /// <returns>a string representation of this sequence of characters.
        /// </returns>
        public override String toString()
        {
            return sb.ToString();
        }

    }

}
